src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -279,46 +286,261 @@ def list_ip_addresses(resource_group_name=None, vm_name=None):
+def create_managed_disk(resource_group_name, disk_name, location=None,
+                        source_blob_uri=None, source_disk=None, source_snapshot=None,
+                        size_gb=None, storage_account_type=None):
+    from azure.mgmt.compute.models import Disk, CreationData, DiskCreateOption, ImageDiskReference
+    location = location or get_resource_group_location(resource_group_name)
+    if source_blob_uri:
+        option = DiskCreateOption.import_enum
+    elif source_disk or source_snapshot:
+        option = DiskCreateOption.copy  # pylint: disable=redefined-variable-type
+    else:
+        option = DiskCreateOption.empty
+
+    creation_data = CreationData(option, source_uri=source_blob_uri,
+                                 image_reference=None,
+                                 source_resource_id=source_disk or source_snapshot)
+
+    if size_gb is None and option == DiskCreateOption.empty:
+        raise CLIError('Please supply size for the empty disk')
 
+    disk = Disk(location, disk_size_gb=size_gb, creation_data=creation_data,
+                account_type=storage_account_type)
+    client = _compute_client_factory()
+    return client.disks.create_or_update(resource_group_name, disk_name, disk)
+
+
+def update_managed_disk(instance, size_gb=None, storage_account_type=None):
+    if size_gb is not None:
+        instance.disk_size_gb = size_gb
+    if storage_account_type is not None:
+        instance.account_type = storage_account_type
+    return instance
+
+
+def attach_managed_data_disk(resource_group_name, vm_name, disk,
+                             new=False, storage_account_type=None, size_gb=None):
+    '''attach a managed disk'''
+    vm = get_vm(resource_group_name, vm_name)
+    from azure.mgmt.compute.models import (CreationData, DiskCreateOptionTypes,
+                                           ManagedDiskParameters, DataDisk)
+    # pylint: disable=no-member
+    luns = [d.lun for d in vm.storage_profile.data_disks] if vm.storage_profile.data_disks else []
+    lun = max(luns) + 1 if luns else 1
+    if new:
+        if not size_gb:
+            raise CLIError('Please provide disk size')
+        data_disk = DataDisk(lun, DiskCreateOptionTypes.empty,
+                             name=parse_resource_id(disk)['name'],
+                             disk_size_gb=size_gb)
+    else:
+        params = ManagedDiskParameters(id=disk,
+                                       storage_account_type=storage_account_type)
+        data_disk = DataDisk(lun, DiskCreateOptionTypes.attach, managed_disk=params)
 
-def attach_existing_disk(resource_group_name, vm_name, vhd, lun=None, disk_name=None, caching=None):
-    ''' Attach an existing disk to an existing Virtual Machine '''
-    return _attach_disk(resource_group_name, vm_name, vhd, DiskCreateOptionTypes.attach,
-                        lun, disk_name, caching)
+    vm.storage_profile.data_disks.append(data_disk)
```

Comment:
the error is pretty good `Disk '/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/yugangwjunk/providers/Microsoft.Compute/disks/d1' cannot be attached as the disk is already owned by VM '/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/yugangwjunk/providers/Microsoft.Compute/virtualMachines/yugangwjunk-1'.`